File: bookstore-backend/app/schemas/order.py
"""Pydantic schemas for the Orders API."""
from typing import Optional

from pydantic import BaseModel


class OrderItemResponse(BaseModel):
    id: str
    book_id: Optional[str]
    book_title: str
    author_name: str
    price_at_purchase: float
    quantity: int
    format: str
    cover_image: Optional[str]
    delivery_date: str

    model_config = {"from_attributes": True}


class OrderResponse(BaseModel):
    id: str
    user_id: str
    subtotal: float
    tax: float
    discount: float
    redeemed_points_amount: int
    total: float
    status: str
    payment_method: str
    coupon_code: Optional[str]
    address_id: Optional[str]
    delivery_date: str          # formatted "Mon, 21 Jul"
    points_awarded: bool
    created_at: str
    items: list[OrderItemResponse]

    model_config = {"from_attributes": True}

File: src/types/order.ts
import type { Book } from './book';

export type OrderStatus = 'Processing' | 'Shipped' | 'Delivered' | 'Cancelled';

/** Legacy client-side order item (uses full Book object). */
export interface OrderItem {
  book: Book;
  quantity: number;
}

/** API-backed order item (snapshot fields only — no full Book). */
export interface ApiOrderItem {
  id: string;
  book_id: string | null;
  book_title: string;
  author_name: string;
  price_at_purchase: number;
  quantity: number;
  format: string;
  cover_image: string | null;
  delivery_date: string;
}

export interface Order {
  id: string;
  userId: string;
  /** Legacy: present for localStorage-created orders */
  items?: OrderItem[];
  /** API-backed order items */
  apiItems?: ApiOrderItem[];
  subtotal: number;
  tax: number;
  discount: number;
  redeemedPointsAmount?: number;
  total: number;
  status: OrderStatus;
  createdAt: number; // Date.now() (ms)
  paymentMethod: string;
  pointsAwarded: boolean;
  deliveryDate?: string;
  pointsEarned?: number;
}
